incubator/binding-kafka-grpc/src/main/java/io/aklivity/zilla/runtime/binding/kafka/grpc/internal/KafkaGrpcBinding.java

```diff
@@ -55,6 +59,15 @@ public int workers(
     public KafkaGrpcBindingContext supply(
         EngineContext context)
     {
-        return new KafkaGrpcBindingContext(config, context);
+        return new KafkaGrpcBindingContext(config, context, this::doSignal);
+    }
+
+    public boolean doSignal(
+        long bindingId)
+    {
+        final AtomicInteger worker = workers.computeIfAbsent(bindingId, id -> new AtomicInteger(0));
+        final int workerNumber = worker.getAndIncrement();
+
+        return workerNumber == 0;
```

Comment:
Suggest changing this to `incrementAndGet` and `workerCount`.

```
        final int workerCount = worker.incrementAndGet();

        return workerCount <= 1;
```

This keeps the idea of a maximum number of workers intact for later.